### DISCUSSION TITLE
Deploying to a google cloud function with typescript

### DISCUSSION THREAD

**Discussion by Sladuca** [prisma/prisma#2106] [Q&A] [answered]
Hello,

I'm looking for some help dealing with an issue when deploying a typescript apollo server using the new prisma client to a google cloud function. For some reason, `prisma generate` isn't being run when it deploys, so I get `Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.` when the function tries to initialize. Obviously `prisma generate` has to run somewhere, but I haven't had much luck getting it to do that. I'm not sure if this is something to do with prisma or something to do with google cloud functions.

Can anyone offer some advice on how to make `prisma-generate` happen when the cloud function deploys? Thank you so much in advance.

Here's my config information

`prisma.schema` generator block
```prisma
generator client {
  provider = "prisma-client-js"
}
```

script I am using to deploy
```sh
export DATABASE_URL="redacted"
yarn compile && yarn prisma2 migrate up --experimental
```

`package.json`
```json
{
  "name": "redacted",
  "version": "1.0.0",
  "main": "dist/index.js",
  "license": "UNLICENSED",
  "dependencies": {
    "@google-cloud/storage": "^4.7.0",
    "@prisma/client": "^2.0.0-beta.1",
    "@types/bcrypt": "^3.0.0",
    "@types/cookie": "^0.3.3",
    "@types/express": "^4.17.3",
    "@types/jsonwebtoken": "^8.3.8",
    "@types/node": "^13.9.2",
    "apollo-server-cloud-functions": "^2.11.0",
    "bcrypt": "^4.0.1",
    "cookie": "^0.4.0",
    "cross-fetch": "^3.0.4",
    "dotenv": "^8.2.0",
    "env-cmd": "^10.1.0",
    "generate-password": "^1.5.1",
    "graphql": "^14.6.0",
    "graphql-scalars": "^1.0.7",
    "graphql-tag": "^2.10.3",
    "graphql-tools": "^4.0.7",
    "jsonwebtoken": "^8.5.1",
    "prisma-client-lib": "^1.34.10"
  },
  "scripts": {
    "compile": "yarn tsc",
    "reset-pg": "sh ./postgres.sh",
    "setup:dev": "source ./env.sh && prisma migrate up --experimental && prisma2 generate",
    "dev": "yarn setup:dev && nodemon -e ts --exec \"yarn compile && functions-framework --target=leto || exit 1\"",
    "predeploy": "yarn prisma2 generate",
    "docs": "graphdoc",
    "deploy": "source ./deploy.sh && gcloud functions deploy leto --project redacted --runtime nodejs10 --trigger-http"
  },
  "graphdoc": {
    "endpoint": "http://localhost:8080",
    "output": "./doc/schema"
  },
  "devDependencies": {
    "@google-cloud/functions-framework": "^1.5.0",
    "@prisma/cli": "^2.0.0-beta.1",
    "@typescript-eslint/eslint-plugin": "^2.25.0",
    "@typescript-eslint/parser": "^2.25.0",
    "eslint": "^6.8.0",
    "faker": "^4.1.0",
    "nodemon": "^2.0.2",
    "typescript": "^3.8.3"
  }
}
```

Full stacktrace of error:

<img width="1200" alt="Screen Shot 2020-04-05 at 11 18 57 PM" src="https://user-images.githubusercontent.com/21142281/78520255-eab74a80-7793-11ea-8ddf-fb117f165b46.png">

**Comment by pantharshit00** [comment, marked as answer]
Hi @Sladuca 

We have a GCP examples in our end to end test. You can refer to that: https://github.com/prisma/prisma2-e2e-tests/tree/master/platforms/gcp-functions

**Reply by Sladuca** [reply]
Thank you!

**Reply by eurobob** [reply]
@pantharshit00 I am using the postinstall script as shown in this example but still experiencing the error described?

**Reply by eurobob** [reply]
Nevermind, if anyone else has this trouble, make sure you're using the right version of typescript. Firebase functions init uses 3.2.2, but prisma requires 3.8.3

**Reply by pantharshit00** [reply]
https://github.com/prisma/prisma/commit/965809658fe8ca31ffc570153c9c8d722a33cb29

We figured out that actual we don't need that specific version of typescript anymore so we have removed the check in the above commit. 

It is available in alpha right now and it will released in the beta.

**Reply by pantharshit00** [reply]
But please make sure you are using a recent version of yarn as that has a bug fix which is required by Prisma

**Reply by jacobakkerboom** [reply]
The link no longer works unfortunately, it seems the gcp-functions folder was moved here: https://github.com/prisma/ecosystem-tests/tree/dev/platforms-serverless/gcp-functions

**Comment by licaibinbin** [comment]
HH

**Comment by Zhomart** [comment]
**UPDATE**: false alarm, use `yarn` instead of npm. You can still ignore node_modules.

I had similar issue, but it was my own fault ignoring `node_modules` in `firebase.json`. Don't add `"**/node_modules/**"` in ignore field:

```json
{
  "functions": {
    "predeploy": [
      "npm --prefix \"$RESOURCE_DIR\" run lint",
      "npm --prefix \"$RESOURCE_DIR\" run build"
    ],
    "source": "functions",
    "ignore": ["**/.*", "*.log", "**/node_modules/**"]
  }
}
```

**Reply by tylim88** [reply]
using yarn works

**Comment by goleary** [comment]
If any poor soul ends up here after struggling to deploy pisma to Firebase Cloud Functions, do yourself a favor and read through this example:

https://github.com/prisma/e2e-tests/blob/dev/platforms-serverless/firebase-functions

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
